### PR TITLE
Extend Advanced CAS

### PR DESCRIPTION
**Pull request by Kexanone**
**When merged this pull request will:**
- Make the CAS module support jets.
- Make the CAS module support the Blackfish VTOL.
- Update module icons for CAS and suppressive fire.
- Create a single function for changing ownership.

### CODE REVIEW

File: @AresModAchillesExpansion/addons/functions_f_achilles/cfgFunctions.hpp [hunk 1]
```diff
@@ -110,6 +121,9 @@ class CfgFunctions
 			class effectFire;
 			class instantBuildingGarrison;
             class changeAccessoires;
+            class advancedPlaneCAS;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/cfgFunctions.hpp]: is the spacing broken on github or not?

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	
+	Parameters:
+		_array		- <ARRAY> The array from which the standard deviation is computed from
+		_ddof		- <INTEGER> [0] The delta degrees of freedom (see description above)
+	
+	Returns:
+		_stdDev		- <SCALAR> The standard deviation
+	
+	Exampes:
+		(begin example)
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: very clear example

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+	Function:
+		Achilles_arrayStdDev
+	
+	Authors:
+		Kex
+	
+	Description:
+		Returns the population standard deviation of the array for _ddof = 0
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: So what does it return then?

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		Returns the sample standard deviation of the array for _ddof = 1
+	
+	Parameters:
+		_array		- <ARRAY> The array from which the standard deviation is computed from
+		_ddof		- <INTEGER> [0] The delta degrees of freedom (see description above)
+	
+	Returns:
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: scalar here, int there

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		_stdDev		- <SCALAR> The standard deviation
+	
+	Exampes:
+		(begin example)
+		(end)
+*/
+params ["_array", ["_ddof", 0, [0]]];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: spacing everywhere please stop pushing it into a oneliner

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_getDirPitchBank.sqf [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		_dirPitchBank	- <ARRAY> List with dir, pitch and bank angle.
+	
+	Exampes:
+		(begin example)
+		[_object] call Achilles_fnc_getDirPitchBank;
+		(end)
+*/
+
+params ["_object"];
+private _vectDir = vectorDirVisual _object;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_getDirPitchBank.sqf]: spacing again

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_transferOwnership.sqf [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+		_object_list	- <ARRAY> of <OBJECT> Objects to transfer
+		_group_list		- <ARRAY> of <GROUP> Groups to transfer
+		_ownerID		- <INTEGER> [clientOwner] The owner ID of the machine the objects and groups should be transfered to
+							By default they are transfered to this machine
+	
+	Returns:
+		nothing
+	
+	Exampes:
+		(begin example)
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_transferOwnership.sqf]: fantastic example

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_transferOwnership.sqf [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+			// reset the unit loadout as soon as they have become local
+			waitUntil {sleep 1; ({not local _x} count _group_list == 0) or {({not isNull _x} count _group_list == 0) or {{{alive _x} count units _x > 0} count _group_list == 0}}};
+			{
+				{
+					private _loadout = _x getVariable ["Achilles_var_tmpLoadout", []];
+					if !(_loadout isEqualTo []) then
+					{
+						_x setUnitLoadout _loadout;
+					};
+					_x setVariable ["Achilles_var_tmpLoadout", nil, true];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_transferOwnership.sqf]: i’d prefer temp

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_vectAngleXY.sqf [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+		// returns 90
+		[[1,0,0],[0,1,0]] call Achilles_fnc_vectAngleXY;
+		(end)
+*/
+
+params
+[
+	["_vecA", [0,0,0], [[]], 3],
+	["_vecB", [0,0,0], [[]], 3]
+];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_vectAngleXY.sqf]: spacing

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_vectAngleXY.sqf [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+_vecB set [2, 0];
+private _angle = acos (_vecA vectorCos _vecB);
+// The sign of det(_vecA_XY _vecB_XY) tells us the orientation of the basis
+private _determinat = (_vecA#0) * (_vecB#1) - (_vecB#0) * (_vecA#1);
+if (_determinat < 0) then
+{
+	_angle = -_angle;
+};
+// return
+_angle
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_vectAngleXY.sqf]: ;

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+											_magIdx		- <INTEGER> Magazine index (derived from Achilles_fnc_getWeaponsMuzzlesMagazines)
+		_offset_custom					- <ARRAY> [0] Target offset in meters that will be added to the weapon's offset
+											Can be used for fine-tuning the targeting
+											E.g. -25 means the helicopter will target 25 m in front of the target.
+	
+	Returns:
+		_success						- <BOOLEAN> True if the CAS run was completed without occurring exception
+	
+	Exampes:
+		(begin example)
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: mmm

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+_aircraft flyInHeight 500;
+_aircraftGroup allowFleeing 0;
+private _prevBehaviour = behaviour _aircraft;
+_aircraft setBehaviour "CARELESS";
+_aircraft disableAI "TARGET";
+_aircraft disableAI "AUTOTARGET";
+
+// get weapon, muzzle, magazine info and unpack it
+_weaponMuzzleMagazineIdx params ["_weapIdx", "_muzzleIdx", "_magIdx"];
+private _weaponsAndMuzzlesAndMagazines = [_aircraft] call Achilles_fnc_getWeaponsMuzzlesMagazines;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: spacing

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+	[_target, _gunner, _muzzle, _magazine, _aircraft, _turretPath] call Achilles_fnc_forceWeaponFire;
+	_i_fire = _i_fire + 1;
+	sleep _reloadTime;
+	_endTime  < time || !canMove _aircraft || !alive driver _aircraft || fuel _aircraft == 0
+};
+if (!canMove _aircraft || !alive driver _aircraft || fuel _aircraft == 0) exitWith {false};
+_aircraft setBehaviour _prevBehaviour;
+_aircraft enableAI "TARGET";
+_aircraft enableAI "AUTOTARGET";
+true
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: ;

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedPlaneCAS.sqf [hunk 14]
```diff
@@ -0,0 +1,388 @@
...
+											_magIdx		- <INTEGER> Magazine index (derived from Achilles_fnc_getWeaponsMuzzlesMagazines)
+		_offset_custom					- <ARRAY> [0] Target offset in meters that will be added to the weapon's offset
+											Can be used for fine-tuning the targeting
+											E.g. -25 means the plane will target 25 m in front of the target.
+	
+	Returns:
+		_success						- <BOOLEAN> True if the CAS run was completed without occurring exception
+	
+	Exampes:
+		(begin example)
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedPlaneCAS.sqf]: mmm

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedPlaneCAS.sqf [hunk 15]
```diff
@@ -0,0 +1,388 @@
...
+
+// Get the initial state of the aircraft
+private _pos_start = _aircraft modelToWorldVisualWorld [0,0,0];
+// The usage of angles instead of vectors is more suitable for trajectories
+([_aircraft] call Achilles_fnc_getDirPitchBank) params ["_dir_start", "_pitch_start", "_bank_start"];
+private _vectDir_start = vectorDirVisual _aircraft;
+// Estimate velocity vector in rendered time, as there is not velocityVisual command
+private _vel_start = _vectDir_start vectorMultiply (vectorMagnitude velocity _aircraft);
+private _speed_start = vectorMagnitude _vel_start;
+private _speedXY_start = vectorMagnitude [_vel_start#0, _vel_start#1, 0];
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedPlaneCAS.sqf]: camelCase everywhere

Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedPlaneCAS.sqf]: Camel case only is a terrible idea for representing math variables. Makes it even more unreadable in my opinion. The underscore emphasizes subscripts.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf (hunk 3 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: Dunno what your problem is.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf (hunk 13 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: Adapted here the CBA convention to not use a semicolon for return values.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf (hunk 3 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: look down one line, why two of them

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf (hunk 2 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_arrayStdDev.sqf]: Added example

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf (hunk 11 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: Added example

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf (hunk 12 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: Dunno what you mean

File: @AresModAchillesExpansion/addons/modules_f_achilles/Objects/functions/fn_ObjectsTransferOwnership.sqf [hunk 16]
```diff
@@ -44,74 +45,7 @@ private _group_list = [];
...
-			{
-				_x setUnitLoadout _loadout;
-			};
-			_x setVariable ["Achilles_var_tmpLoadout", nil, true];
-		} forEach units _x;
-	} forEach _group_list;
-};
-[localize "STR_AMAE_TRANSFER_TO" + " " + (_options select _owner)] call Ares_fnc_ShowZeusMessage;
+[_object_list, _group_list, _ownerID] call Achilles_fnc_transferOwnership;
+[localize "STR_AMAE_TRANSFER_TO" + " " + (_options select _selection)] call Ares_fnc_ShowZeusMessage;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_achilles/Objects/functions/fn_ObjectsTransferOwnership.sqf]: Why isn't this using `format`?

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf (hunk 13 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/features/fn_advancedBlackfishCAS.sqf]: https://github.com/ArmaAchilles/Achilles/blob/AddonRestructuring/conventions.txt

File: @AresModAchillesExpansion/addons/modules_f_achilles/Objects/functions/fn_ObjectsTransferOwnership.sqf (hunk 16 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/modules_f_achilles/Objects/functions/fn_ObjectsTransferOwnership.sqf]: I will use `joinString`.

**Pull request merged**